### ISSUE TITLE
Merge migration generator into dtk-tools

### ISSUE THREAD

**Issue by ckirkman-IDM** [InstituteforDiseaseModeling/dtk-tools-malaria#61]
- [ ] Migration generator merged into the dtk-tools core
- [ ] Tested
- [ ] Documented

**Comment by devclinton** [comment]
Nearly done. Currently writing basic unittests

**Comment by devclinton** [comment]
https://github.com/InstituteforDiseaseModeling/dtk-tools/pull/1653

**Comment by YeChen-IDM** [comment]
Bugs found and fixes proposed:

1. **MigrationFile.py:**

      - update limit for max destinations to 100 for any migration type.

     - In migration header, DatavalueCount should be the actual max destinations for a single node not the limit for max destinations(MAX_DESTINATIONS_BY_ROUTE)

     -  When compiled_demographics_file_path is not used, offset_str should use capital letters as the format instead of lower case letters.

     - use with open() to avoid ResourceWarning.

2. Add DeprecationWarning in **GravityModelRatesGenerator.py, GeoGraphGenerator and SmallWorldGridGraphGenerator**

3. **StaticGravityModelRatesGenerator.py:**

     - add check for gravity_params to make sure migration rate is proportional to population and inversely proportional to distance.

     - add check for demographics file.

     - add more details in comments.

4. **MigrationGenerator.py**

     - better handling for migration_file_name input.

     - add checks for migration_type and link_rates_model arguments.

     - in generate_migration(), if user pass in demographics_file_path, use it the generate the migration header. If not, requires a new argument idRef to generate the header. This is required by Emod.
 If both are not provided, raise ValueError.

**Comment by YeChen-IDM** [comment]
fix is here: https://github.com/YeChen-IDM/dtk-tools/commit/a3d0473287021a600f0c90995082cf938d1a9b14.
I am working on adding tests.

**Comment by YeChen-IDM** [comment]
tests and update will be merged to dtk-tools with PR: https://github.com/InstituteforDiseaseModeling/dtk-tools/pull/1681

**Comment by YeChen-IDM** [comment]
waiting on PR: https://github.com/InstituteforDiseaseModeling/dtk-tools/pull/1681 to close this ticket.

**Comment by YeChen-IDM** [comment]
Fix and tests are merged to dtk-tools master. pass this ticket to doc team.
